lab-davide/server.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const http = require('http');
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const cowsay = require('cowsay');
+
+const port = process.argv[2] || 3000;
+
+
+const server = http.createServer(function(req, res){
+  req.url =  parseUrl(req.url);
+  req.url.query = parseQuery(req.url.query);
+
+  console.log(req.url.query);
+  console.log(req.url.path);
+  console.log(req.url);
+  console.log(req.url.query.text);
+  if(req.url.pathname == '/cowsay' && req.method === 'GET' && req.url.query.text) {
+    console.log('hello');
+    var message = cowsay.say({text: req.url.query.text});
+    res.writeHead(200, { 'Content-Type': 'text/plain' });
+    res.write(message);
+    res.end();
+    return;
+  }
+
+  if(req.url.pathname === '/cowsay' && req.method === 'POST') {
```

Comment:
What happens if both of these are true but there is no request body? It's supposed to return a 400, but it wont.